pallets/pallet-lottery/src/lib.rs

```diff
@@ -0,0 +1,1077 @@
+        #[pallet::call_index(3)]
+        #[pallet::weight(0)]
+        pub fn rebalance_stake(origin: OriginFor<T>) -> DispatchResult {
+            T::ManageOrigin::ensure_origin(origin.clone())?;
+            Err(crate::pallet::DispatchError::Other(
+                Error::<T>::NotImplemented.into(),
+            ))
+
+            // withdraw from overallocated collators, wait until funds unlock, re-allocate to underallocated collators
+            // TODO: find some balancing algorithm that does this or just reuse the one we use for deposits
+
+            // Self::deposit_event(Event::StartedRebalance(amount));
+            // Ok(())
+        }
+
+        /// Starts the lottery by scheduling a [`Call::draw_lottery`] call
+        ///
+        /// Can only be called by the account set as [`Config::ManageOrigin`]
+        ///
+        /// # Errors
+        ///
+        /// Returns an error if:
+        /// * BadOrigin: Caller is not ManageOrigin
+        /// * The pallet does not have enough funds to pay for gas fees for at least the first drawing.
+        /// * The drawing interval is zero or negative.
+        /// * The Scheduler implementation failed to schedule the [`Call::draw_lottery`] call.
+        ///
+        /// # Details
+        ///
+        /// This function schedules a [`Call::draw_lottery`] call with a delay specified by the [`Config::DrawingInterval`] configuration
+        /// using [`frame_support::traits::schedule::Named`] with the lottery pallet's pallet ID configured with [`Config::LotteryPot`] as identifier.
+        /// If the lottery is already started, this function will fail.
+        ///
+        /// You can always learn what block the next drawing - if any - will happen by calling [`Self::next_drawing_at`]
+        #[pallet::call_index(4)]
+        #[pallet::weight(<T as Config>::WeightInfo::start_lottery())]
+        pub fn start_lottery(origin: OriginFor<T>) -> DispatchResult {
+            T::ManageOrigin::ensure_origin(origin.clone())?;
+            ensure!(
+                Self::next_drawing_at().is_none(),
+                Error::<T>::LotteryIsRunning
+            );
+            // Pallet has enough funds to pay gas fees for at least the first drawing
+            ensure!(
+                Self::surplus_funds() >= Self::gas_reserve(),
+                Error::<T>::PotBalanceBelowGasReserve
+            );
+            // NOTE: If more than gas_reserve is in the pallet, the full excess will be paid out to the winner of the next drawing! This is intended to dope the winning balance with extra rewards
+
+            let drawing_interval = <T as Config>::DrawingInterval::get();
+            ensure!(
+                drawing_interval > 0u32.into(),
+                Error::<T>::PalletMisconfigured
+            );
+            let lottery_drawing_call: CallOf<T> = Call::draw_lottery {}.into();
+            T::Scheduler::schedule_named(
+                Self::lottery_schedule_id(),
```

Comment:
I'm not sure we should use the same name for every scheduled task, because if we use the same name, the previous scheduled task storage will be overwritten.

Reply:
there can not be multiple lotteries running in parallel. This usecase is simply not supported for now

Reply:
Can we add this as a comment